### PR TITLE
Aggressive parallel test execution [testing]

### PR DESCRIPTION
**Pull request by yschimke**
Testing for now

### CODE REVIEW

**Comment by yschimke** [comment]
I suspect travis is only giving us so no clear wins here.  But locally runs in 90 seconds e.g.

https://scans.gradle.com/s/zhlsqk5azvs3q

File: src/test/resources/junit-platform.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+junit.jupiter.execution.parallel.enabled=true
+junit.jupiter.execution.parallel.mode.default=concurrent
```
Comment (swankjesse) [inline src/test/resources/junit-platform.properties]: Is JUnit 5 involved?

Comment (yschimke) [reply src/test/resources/junit-platform.properties]: There are 3 potential things here

1) Parallel Modules
2) Forking Multiple JVMs per module test run and dividing Test Cases between them
3) Concurrency within a single VM - https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-parallel-execution

Junit 5 runner is relevant for the 3rd, but shouldn't affect the test classes, just a different test runner.  I may not have unlocked this yet...

Comment (yschimke) [reply src/test/resources/junit-platform.properties]: I'm not in a rush until I confirm all 3 are active and see if results are ok.

**Comment by yschimke** [comment]
Flaky tests when run in parallel mode

https://circleci.com/build-insights/gh/square/okhttp/pull%252F4737

**Pull request closed without merging**